Admin UI can't create event with illegal XML characters in its meta data

**Describe the bug**
A clear and concise description of what you see:
*When I supply invalid XML (e.g. a title containing a vertical tab character) for the event meta data while uploading a video in the Admin UI I get an error message: `The event could not be created` instead of a new event in the list of events*

**To Reproduce**
Steps to reproduce the behavior:
1. Go to https://stable.opencast.org
2. Click on 'Add event'
3. Execute the following code in your JS console (GitHub replaces the vertical tab with a unicode replacement char):
```javascript
$('#add-event-modal .ng-scope.active td').eq(1).find('i:first').click().parents('td').find('input').focus().val('A title courtesy by Felix, containing' + String.fromCharCode(0x0b) + 'a vertical tab.').trigger('change');
```
4. Click into the input field that was filled, click somewhere else
5. Proceed, supply a file and a workflow finally click `Create`
6. The event is uploaded. Upon upload completes, there is a message `The event could not be created`

**Environment Information**
 - Browser Firefox 84.0 (64-bit)
 - Opencast version with minor release Opencast 9.0.0.SNAPSHOT - c3e94df (r/9.x) but also reproducible with 8.7.0

**Video**
<video src="https://user-images.githubusercontent.com/69673133/103480663-cf867380-4dd5-11eb-9ca7-0b2720d8dc33.mp4">

**Requests**

The request to https://stable.opencast.org/admin-ng/event/new ended with a 500 error with no response body.

**Logs**
```
2021-01-03T14:37:22,017 | ERROR | (DublinCoreUtil:100) - Unable to load metadata from catalog 'https://stable.opencast.org/files/mediapackage/1a79d832-31af-4d8a-a629-0903ef06a3e7/baef0926-d018-45f2-a4d6-f560fb668849/dublincore.xml'
java.lang.RuntimeException: Unable to read DublinCore catalog, XML parsing failed.
	at org.opencastproject.metadata.dublincore.DublinCores.read(DublinCores.java:187) ~[?:?]
	at org.opencastproject.metadata.dublincore.DublinCoreUtil.loadDublinCore(DublinCoreUtil.java:98) ~[?:?]
	at org.opencastproject.metadata.dublincore.DublinCoreUtil$1.apply(DublinCoreUtil.java:83) ~[?:?]
	at org.opencastproject.metadata.dublincore.DublinCoreUtil$1.apply(DublinCoreUtil.java:80) ~[?:?]
	at com.entwinemedia.fn.data.Opt.fmap(Opt.java:94) ~[?:?]
	at com.entwinemedia.fn.data.Opt.map(Opt.java:99) ~[?:?]
	at org.opencastproject.metadata.dublincore.DublinCoreUtil.loadDublinCore(DublinCoreUtil.java:80) ~[?:?]
	at org.opencastproject.metadata.dublincore.DublinCoreUtil.loadEpisodeDublinCore(DublinCoreUtil.java:70) ~[?:?]
	at org.opencastproject.index.service.impl.IndexServiceImpl.getDublinCoreCatalog(IndexServiceImpl.java:1039) ~[?:?]
	at org.opencastproject.index.service.impl.IndexServiceImpl.createEvent(IndexServiceImpl.java:930) ~[?:?]
	at org.opencastproject.index.service.impl.IndexServiceImpl.createEvent(IndexServiceImpl.java:841) ~[?:?]
	at org.opencastproject.index.service.impl.IndexServiceImpl.createEvent(IndexServiceImpl.java:595) ~[?:?]
	at org.opencastproject.adminui.endpoint.AbstractEventEndpoint.createNewEvent(AbstractEventEndpoint.java:2256) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_265]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_265]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_265]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_265]
	at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179) ~[?:?]
	at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96) ~[?:?]
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:201) ~[?:?]
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:104) ~[?:?]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59) ~[?:?]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96) ~[?:?]
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308) ~[?:?]
	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121) ~[?:?]
	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:267) ~[?:?]
	at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234) ~[?:?]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208) ~[?:?]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160) ~[?:?]
	at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:216) ~[?:?]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:301) ~[?:?]
	at org.opencastproject.kernel.rest.RestPublisher$RestServlet.handleRequest(RestPublisher.java:558) ~[?:?]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doPost(AbstractHTTPServlet.java:220) ~[?:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707) ~[!/:3.1.0]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:276) ~[?:?]
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:763) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1631) ~[!/:9.4.28.v20200408]
	at org.opencastproject.kernel.rest.JsonpFilter.doFilter(JsonpFilter.java:113) ~[?:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) ~[!/:9.4.28.v20200408]
	at org.opencastproject.kernel.filter.proxy.TransparentProxyFilter.doFilter(TransparentProxyFilter.java:63) ~[?:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) ~[!/:9.4.28.v20200408]
	at org.opencastproject.security.urlsigning.filter.UrlSigningFilter.doFilter(UrlSigningFilter.java:86) ~[!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) ~[!/:9.4.28.v20200408]
	at org.opencastproject.kernel.security.RemoteUserAndOrganizationFilter.doFilter(RemoteUserAndOrganizationFilter.java:247) ~[?:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) ~[!/:9.4.28.v20200408]
	at org.opencastproject.kernel.rest.CurrentJobFilter.doFilter(CurrentJobFilter.java:87) ~[?:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) ~[!/:9.4.28.v20200408]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330) ~[?:?]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:118) ~[?:?]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:84) ~[?:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) ~[?:?]
	at org.opencastproject.kernel.security.AsyncTimeoutRedirectFilter.doFilter(AsyncTimeoutRedirectFilter.java:60) ~[?:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) ~[?:?]
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:113) ~[?:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) ~[?:?]
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:103) ~[?:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) ~[?:?]
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:113) ~[?:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) ~[?:?]
	at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:146) ~[?:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) ~[?:?]
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:54) ~[?:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) ~[?:?]
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:45) ~[?:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) ~[?:?]
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:98) ~[?:?]
	at org.springframework.security.oauth.provider.filter.OAuthProviderProcessingFilter.doFilter(OAuthProviderProcessingFilter.java:182) ~[?:?]
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:102) ~[?:?]
	at org.springframework.security.web.authentication.www.BasicAuthenticationFilter.doFilter(BasicAuthenticationFilter.java:150) ~[?:?]
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:102) ~[?:?]
	at org.springframework.security.web.authentication.www.DigestAuthenticationFilter.doFilter(DigestAuthenticationFilter.java:115) ~[?:?]
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:102) ~[?:?]
	at org.springframework.web.filter.CompositeFilter.doFilter(CompositeFilter.java:82) ~[?:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) ~[?:?]
	at org.springframework.security.web.authentication.ui.DefaultLoginPageGeneratingFilter.doFilter(DefaultLoginPageGeneratingFilter.java:91) ~[?:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) ~[?:?]
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:183) ~[?:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) ~[?:?]
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:105) ~[?:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) ~[?:?]
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:87) ~[?:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) ~[?:?]
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:192) ~[?:?]
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:160) ~[?:?]
	at org.opencastproject.kernel.security.SecurityFilter.doFilter(SecurityFilter.java:124) ~[?:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) ~[!/:9.4.28.v20200408]
	at org.opencastproject.kernel.security.OrganizationFilter.doFilter(OrganizationFilter.java:135) ~[?:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) ~[!/:9.4.28.v20200408]
	at org.opencastproject.kernel.filter.https.HttpsFilter.doFilter(HttpsFilter.java:73) ~[?:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) ~[!/:9.4.28.v20200408]
	at org.opencastproject.kernel.rest.CleanSessionsFilter.doFilter(CleanSessionsFilter.java:89) ~[?:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:226) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:549) ~[!/:9.4.28.v20200408]
	at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:71) ~[!/:?]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:602) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1610) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1363) ~[!/:9.4.28.v20200408]
	at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:298) ~[!/:?]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:489) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1580) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1278) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) ~[!/:9.4.28.v20200408]
	at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:82) ~[!/:?]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.Server.handle(Server.java:500) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383) ~[!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:547) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938) [!/:9.4.28.v20200408]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_265]
Caused by: org.xml.sax.SAXParseException: Character reference "&#11" is an invalid XML character.
	at org.apache.xerces.util.ErrorHandlerWrapper.createSAXParseException(Unknown Source) ~[?:?]
	at org.apache.xerces.util.ErrorHandlerWrapper.fatalError(Unknown Source) ~[?:?]
	at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source) ~[?:?]
	at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source) ~[?:?]
	at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source) ~[?:?]
	at org.apache.xerces.impl.XMLScanner.reportFatalError(Unknown Source) ~[?:?]
	at org.apache.xerces.impl.XMLScanner.scanCharReferenceValue(Unknown Source) ~[?:?]
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanCharReference(Unknown Source) ~[?:?]
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch(Unknown Source) ~[?:?]
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source) ~[?:?]
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source) ~[?:?]
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source) ~[?:?]
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source) ~[?:?]
	at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source) ~[?:?]
	at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source) ~[?:?]
	at org.apache.xerces.jaxp.SAXParserImpl.parse(Unknown Source) ~[?:?]
	at org.opencastproject.metadata.dublincore.DublinCoreXmlFormat.readImpl(DublinCoreXmlFormat.java:247) ~[?:?]
	at org.opencastproject.metadata.dublincore.DublinCoreXmlFormat.read(DublinCoreXmlFormat.java:145) ~[?:?]
	at org.opencastproject.metadata.dublincore.DublinCores.read(DublinCores.java:185) ~[?:?]
	... 127 more
```

**Notes**

Vertical tab characters regularly get into Opencast by Copy & Paste from Powerpoint slides.

--- a/modules/admin-ui-frontend/app/index.html
+++ b/modules/admin-ui-frontend/app/index.html
@@ -298,6 +298,7 @@
     <script src="scripts/shared/directives/selectBoxDirective.js"></script>
     <script src="scripts/shared/directives/navDirective.js"></script>
     <script src="scripts/shared/directives/tableDirective.js"></script>
+    <script src="scripts/shared/directives/sanitizeXMLDirective.js"></script>
     <script src="scripts/shared/directives/statsDirective.js"></script>
     <script src="scripts/shared/directives/collapsibleBoxDirective.js"></script>
     <script src="scripts/shared/directives/confirmationModalDirective.js"></script>

--- a/modules/admin-ui-frontend/app/scripts/modules/events/controllers/eventController.js
+++ b/modules/admin-ui-frontend/app/scripts/modules/events/controllers/eventController.js
@@ -724,6 +724,7 @@ angular.module('adminNg.controllers')
       if (Object.prototype.hasOwnProperty.call(catalog, 'fields')) {
         for (var fieldNo in catalog.fields) {
           var field = catalog.fields[fieldNo];
+
           if (Object.prototype.hasOwnProperty.call(field, 'collection')) {
             field.collection = [];
           }

--- a/modules/admin-ui-frontend/app/scripts/modules/events/controllers/serieController.js
+++ b/modules/admin-ui-frontend/app/scripts/modules/events/controllers/serieController.js
@@ -312,6 +312,7 @@ angular.module('adminNg.controllers')
       if (Object.prototype.hasOwnProperty.call(catalog, 'fields')) {
         for (var fieldNo in catalog.fields) {
           var field = catalog.fields[fieldNo];
+
           if (Object.prototype.hasOwnProperty.call(field, 'collection')) {
             field.collection = [];
           }

--- a/modules/admin-ui-frontend/app/scripts/shared/directives/sanitizeXMLDirective.js
+++ b/modules/admin-ui-frontend/app/scripts/shared/directives/sanitizeXMLDirective.js
@@ -1,0 +1,67 @@
+/**
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+'use strict';
+
+/**
+ * @ngdoc directive
+ * @name adminNg.modal.sanitizeXml
+ * @description
+ * Removes illegal XML characters from the input string.
+ *
+ * Illegal characters are all characters not in the following list:
+ * x09: Character Tabulation
+ * x0A: Line Feed
+ * x0D: Carriage Return
+ * x20-xD7FF
+ * uE000-uFFFD
+ * x10000-x10FFFF
+ *
+ * Usage: Set this directive in the input element that needs to be sanitized.
+ *
+ * @example
+ * <input sanitize-xml="user.username"/>
+ */
+angular.module('adminNg.directives')
+.directive('sanitizeXml', function () {
+  return {
+    require: 'ngModel',
+    link: function (scope, elem, attrs, ctrl) {
+      scope.unregisterWatch = scope.$watch(attrs.sanitizeXml, function (input) {
+        // Regexp matches illegal characters.
+        // Unfortunately javascript does not play nice with the higher planes of unicode (x10000-x10FFFF).
+        // With ES6 we might be able to switch to a nicer expression like
+        // /([^\x09\x0A\x0D\x20-\uD7FF\uE000-\uFFFC\u{10000}-\u{10FFFF}])/ug
+        // eslint-disable-next-line no-control-regex
+        var NOT_SAFE_IN_XML_1_0 = new RegExp('((?:[\0-\x08\x0B\f\x0E-\x1F\uFFFD\uFFFE\uFFFF]|'
+          + '[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFF]))', 'g');
+
+        if (typeof input === 'string') {
+          ctrl.$setViewValue(input.replace(NOT_SAFE_IN_XML_1_0, ''));
+          ctrl.$render();
+        }
+      });
+
+      scope.$on('$destroy', function () {
+        scope.unregisterWatch();
+      });
+    }
+  };
+});

--- a/modules/admin-ui-frontend/app/scripts/shared/partials/modals/acl-details.html
+++ b/modules/admin-ui-frontend/app/scripts/shared/partials/modals/acl-details.html
@@ -39,7 +39,8 @@
                         ng-model="metadata.name"
                         ng-model-options="{ debounce: {'default': 500} }"
                         ng-required="true"
-                        placeholder="{{ 'USERS.ACLS.DETAILS.METADATA.NAME.PLACEHOLDER' | translate }}">
+                        placeholder="{{ 'USERS.ACLS.DETAILS.METADATA.NAME.PLACEHOLDER' | translate }}"
+                        sanitize-xml="metadata.name">
                     </td>
                   </tr>
                 </table>

--- a/modules/admin-ui-frontend/app/scripts/shared/partials/modals/event-details.html
+++ b/modules/admin-ui-frontend/app/scripts/shared/partials/modals/event-details.html
@@ -100,7 +100,8 @@
                 </td>
                 <td admin-ng-editable
                     required-role="ROLE_UI_EVENTS_DETAILS_METADATA_EDIT"
-                    params="row" save="getSaveFunction(episodeCatalog.flavor)">
+                    params="row" save="getSaveFunction(episodeCatalog.flavor)"
+                    sanitize-xml="row.value" ng-model="row.value">
                 </td>
               </tr>
             </table>
@@ -126,7 +127,8 @@
                 </td>
                 <td admin-ng-editable
                     required-role="ROLE_UI_EVENTS_DETAILS_METADATA_EDIT"
-                    params="row" save="getSaveFunction(catalog.flavor)">
+                    params="row" save="getSaveFunction(catalog.flavor)"
+                    sanitize-xml="row.value" ng-model="row.value">
                 </td>
               </tr>
             </table>

--- a/modules/admin-ui-frontend/app/scripts/shared/partials/modals/group-modal.html
+++ b/modules/admin-ui-frontend/app/scripts/shared/partials/modals/group-modal.html
@@ -32,11 +32,17 @@
                    placeholder="{{ 'USERS.GROUPS.DETAILS.FORM.NAME' | translate }}..."
                    name="name"
                    ng-class="blurred.name == true && !groupForm.name.$valid ? 'error' : ''"
-                   ng-blur="blurred.name=true">
+                   ng-blur="blurred.name=true"
+                   sanitize-xml="group.name">
           </div>
           <div class="row">
             <label translate="USERS.GROUPS.DETAILS.FORM.DESCRIPTION"><!-- Description --></label>
-            <textarea ng-model="group.description" placeholder="Description..." rows="4" name="description"></textarea>
+            <textarea ng-model="group.description"
+                      placeholder="Description..."
+                      rows="4"
+                      name="description"
+                      sanitize-xml="group.description">
+            </textarea>
           </div>
         </div>
       </div>

--- a/modules/admin-ui-frontend/app/scripts/shared/partials/modals/series-details.html
+++ b/modules/admin-ui-frontend/app/scripts/shared/partials/modals/series-details.html
@@ -86,7 +86,8 @@
                 </td>
                 <td admin-ng-editable
                     required-role="ROLE_UI_SERIES_DETAILS_METADATA_EDIT"
-                    params="row" save="getSaveFunction(seriesCatalog.flavor)">
+                    params="row" save="getSaveFunction(seriesCatalog.flavor)"
+                    sanitize-xml="row.value" ng-model="row.value">
                 </td>
               </tr>
             </table>
@@ -112,7 +113,8 @@
                 </td>
                 <td admin-ng-editable
                     required-role="ROLE_UI_SERIES_DETAILS_METADATA_EDIT"
-                    params="row" save="getSaveFunction(catalog.flavor)">
+                    params="row" save="getSaveFunction(catalog.flavor)"
+                    sanitize-xml="row.value" ng-model="row.value">
                 </td>
               </tr>
             </table>

--- a/modules/admin-ui-frontend/app/scripts/shared/partials/modals/user-modal.html
+++ b/modules/admin-ui-frontend/app/scripts/shared/partials/modals/user-modal.html
@@ -51,6 +51,7 @@
                    name="username"
                    required
                    placeholder="{{ 'USERS.USERS.DETAILS.FORM.USERNAME' | translate }}..."
+                   sanitize-xml="user.username"
                    ng-disabled="action == 'edit'"
                    ng-init="blurred.username = false"
                    ng-model="user.username"
@@ -64,6 +65,7 @@
                    name="name"
                    required
                    placeholder="{{ 'USERS.USERS.DETAILS.FORM.NAME' | translate }}..."
+                   sanitize-xml="user.name"
                    ng-disabled="!manageable"
                    ng-init="blurred.name = false"
                    ng-model="user.name"

--- a/modules/admin-ui-frontend/app/scripts/shared/partials/wizards/new-acl.html
+++ b/modules/admin-ui-frontend/app/scripts/shared/partials/wizards/new-acl.html
@@ -14,6 +14,7 @@
                       class="hidden-input"
                       tabindex="1"
                       focushere
+                      sanitize-xml="wizard.step.metadata.name"
                       ng-model="wizard.step.metadata.name"
                       placeholder="{{ 'USERS.ACLS.NEW.METADATA.NAME.PLACEHOLDER' | translate }}">
                   </td>

--- a/modules/admin-ui-frontend/app/scripts/shared/partials/wizards/new-event.html
+++ b/modules/admin-ui-frontend/app/scripts/shared/partials/wizards/new-event.html
@@ -16,7 +16,8 @@
                 <td admin-ng-editable
                     class="editable" name="row.id"
                                      target="dublincore/episode"
-                                     params="row" save="$parent.wizard.save">
+                                     params="row" save="$parent.wizard.save"
+                                     sanitize-xml="row.value" ng-model="row.value">
                 </td>
               </tr>
             </table>
@@ -40,7 +41,8 @@
                 <td>{{ row.label | translate }} <i ng-show="row.required" class="required">*</i></td>
                 <td admin-ng-editable
                     class="editable" name="row.id" target="{{name}}"
-                                                   params="row" save="$parent.wizard.save">
+                                                   params="row" save="$parent.wizard.save"
+                                                   sanitize-xml="row.value" ng-model="row.value">
                 </td>
               </tr>
             </table>

--- a/modules/admin-ui-frontend/app/scripts/shared/partials/wizards/new-group.html
+++ b/modules/admin-ui-frontend/app/scripts/shared/partials/wizards/new-group.html
@@ -5,14 +5,14 @@
       <div class="form-container">
         <div class="row">
           <label translate="USERS.GROUPS.DETAILS.FORM.NAME" class="required"><!-- Group Name --></label>
-          <input ng-init="blurred.name = false" ng-model="wizard.step.metadata.name" type="text" ng-required="true"
+          <input ng-init="blurred.name = false" ng-model="wizard.step.metadata.name" type="text" ng-required="true" sanitize-xml="wizard.step.metadata.name"
                                                                                                  placeholder="{{ 'USERS.GROUPS.DETAILS.FORM.NAME' | translate }}..." name="name"
                                                                                                                                                                      ng-class="blurred.name == true && !groupForm.name.$valid ? 'error' : ''" ng-blur="blurred.name=true"
                                                                                                                                                                                                                                               tabindex="1" focushere>
         </div>
         <div class="row">
           <label translate="USERS.GROUPS.DETAILS.FORM.DESCRIPTION"><!-- Group Description --></label>
-          <textarea ng-model="wizard.step.metadata.description"
+          <textarea ng-model="wizard.step.metadata.description" sanitize-xml="wizard.step.metadata.description"
                     placeholder="{{ 'USERS.GROUPS.DETAILS.FORM.DESCRIPTION' | translate }}..." rows="4"
                                                                                                name="description" tabindex="2"></textarea>
         </div>

--- a/modules/admin-ui-frontend/app/scripts/shared/partials/wizards/new-series.html
+++ b/modules/admin-ui-frontend/app/scripts/shared/partials/wizards/new-series.html
@@ -11,7 +11,8 @@
                 <td admin-ng-editable
                     class="editable"
                     name="row.id"
-                    params="row" save="$parent.wizard.save">
+                    params="row" save="$parent.wizard.save"
+                    sanitize-xml="row.value" ng-model="row.value">
                 </td>
               </tr>
             </table>
@@ -36,7 +37,8 @@
                     class="editable"
                     name="row.id"
                     target="{{name}}"
-                    params="row" save="$parent.wizard.save">
+                    params="row" save="$parent.wizard.save"
+                    sanitize-xml="row.value" ng-model="row.value">
                 </td>
               </tr>
             </table>

--- a/modules/admin-ui-frontend/app/scripts/shared/services/jsHelperService.js
+++ b/modules/admin-ui-frontend/app/scripts/shared/services/jsHelperService.js
@@ -398,7 +398,8 @@ angular.module('adminNg.services')
             }
           });
         });
-      }
+      },
+
     };
   }
 ]);

--- a/modules/admin-ui-frontend/app/scripts/shared/services/wizards/new-event/metadataExtended.js
+++ b/modules/admin-ui-frontend/app/scripts/shared/services/wizards/new-event/metadataExtended.js
@@ -21,137 +21,139 @@
 'use strict';
 
 angular.module('adminNg.services')
-.factory('NewEventMetadataExtended', ['NewEventMetadataResource', function (NewEventMetadataResource) {
-  var MetadataExtended = function () {
-    var me = this, i;
-    this.ud = {};
-    this.requiredMetadata = {};
+.factory('NewEventMetadataExtended', ['NewEventMetadataResource',
+  function (NewEventMetadataResource) {
+    var MetadataExtended = function () {
+      var me = this, i;
+      this.ud = {};
+      this.requiredMetadata = {};
 
-    me.isMetadataExtendedState = true;
+      me.isMetadataExtendedState = true;
 
-    this.updateRequiredMetadata = function(fieldId, value) {
-      if (angular.isDefined(value) && value.length > 0) {
-        me.requiredMetadata[fieldId] = true;
-      } else {
-        me.requiredMetadata[fieldId] = false;
-      }
-    };
-
-    // As soon as the required metadata fields arrive from the backend,
-    // we check which are mandatory.
-    // This information will be needed in order to tell if we can move
-    // on to the next page of the wizard.
-    this.postProcessMetadata = function (data) {
-      var fields = [], chunk;
-      for (chunk in data) {
-        if (Object.prototype.hasOwnProperty.call(data, chunk)) {
-          // extended metadata is every object in the returned data which
-          // does not start with a dollar sign and which isn't dublincore/episode
-          if (chunk !== 'dublincore/episode' && chunk.charAt(0) !== '$') {
-            me.ud[chunk] = {fields: data[chunk].fields};
-            me.ud[chunk].flavor = data[chunk].flavor;
-            me.ud[chunk].title = data[chunk].title;
-            fields = fields.concat(data[chunk].fields);
-          }
+      this.updateRequiredMetadata = function(fieldId, value) {
+        if (angular.isDefined(value) && value.length > 0) {
+          me.requiredMetadata[fieldId] = true;
+        } else {
+          me.requiredMetadata[fieldId] = false;
         }
-      }
-      // we go for the extended metadata here
-      if (fields.length > 0) {
-        for (i = 0; i < fields.length; i++) {
-          // just hooking the tab index up here, as this is already running through all elements
-          fields[i].tabindex = i + 1;
-          if (fields[i].required) {
-            me.updateRequiredMetadata(fields[i].id, fields[i].value);
-            if (fields[i].type === 'boolean') {
-              // set all boolean fields to false by default
-              fields[i].value = false;
-              // remark: since booleans are always present, we
-              // do not add this property to the required fields.
+      };
+
+      // As soon as the required metadata fields arrive from the backend,
+      // we check which are mandatory.
+      // This information will be needed in order to tell if we can move
+      // on to the next page of the wizard.
+      this.postProcessMetadata = function (data) {
+        var fields = [], chunk;
+        for (chunk in data) {
+          if (Object.prototype.hasOwnProperty.call(data, chunk)) {
+            // extended metadata is every object in the returned data which
+            // does not start with a dollar sign and which isn't dublincore/episode
+            if (chunk !== 'dublincore/episode' && chunk.charAt(0) !== '$') {
+              me.ud[chunk] = {fields: data[chunk].fields};
+              me.ud[chunk].flavor = data[chunk].flavor;
+              me.ud[chunk].title = data[chunk].title;
+              fields = fields.concat(data[chunk].fields);
             }
           }
         }
-        me.visible = true;
-      }
-      else {
-        me.visible = false;
-      }
-    };
-
-    this.metadata = NewEventMetadataResource.get(this.postProcessMetadata);
-
-    // Checks if the current state of this wizard is valid and we are
-    // ready to move on.
-    this.isValid = function () {
-      var result = true;
-      //FIXME: The angular validation should rather be used,
-      // unfortunately it didn't work in this context.
-      angular.forEach(me.requiredMetadata, function (item) {
-        if (item === false) {
-          result = false;
+        // we go for the extended metadata here
+        if (fields.length > 0) {
+          for (i = 0; i < fields.length; i++) {
+            // just hooking the tab index up here, as this is already running through all elements
+            fields[i].tabindex = i + 1;
+            if (fields[i].required) {
+              me.updateRequiredMetadata(fields[i].id, fields[i].value);
+              if (fields[i].type === 'boolean') {
+                // set all boolean fields to false by default
+                fields[i].value = false;
+                // remark: since booleans are always present, we
+                // do not add this property to the required fields.
+              }
+            }
+          }
+          me.visible = true;
         }
-      });
-      return result;
-    };
+        else {
+          me.visible = false;
+        }
+      };
 
-    this.save = function (scope) {
-      //FIXME: This should be nicer, rather propagate the id and values instead of looking for them in the parent scope.
-      var field = scope.$parent.params,
-          target = scope.$parent.target;
+      this.metadata = NewEventMetadataResource.get(this.postProcessMetadata);
 
-      if (field.collection) {
-        if (angular.isArray(field.value)) {
-          field.presentableValue = field.value;
+      // Checks if the current state of this wizard is valid and we are
+      // ready to move on.
+      this.isValid = function () {
+        var result = true;
+        //FIXME: The angular validation should rather be used,
+        // unfortunately it didn't work in this context.
+        angular.forEach(me.requiredMetadata, function (item) {
+          if (item === false) {
+            result = false;
+          }
+        });
+        return result;
+      };
+
+      this.save = function (scope) {
+        //FIXME: This should be nicer,
+        //rather propagate the id and values instead of looking for them in the parent scope.
+        var field = scope.$parent.params,
+            target = scope.$parent.target;
+
+        if (field.collection) {
+          if (angular.isArray(field.value)) {
+            field.presentableValue = field.value;
+          } else {
+            field.presentableValue = field.collection[field.value];
+          }
         } else {
-          field.presentableValue = field.collection[field.value];
+          field.presentableValue = field.value;
         }
-      } else {
-        field.presentableValue = field.value;
-      }
 
-      if (angular.isDefined(me.requiredMetadata[field.id])) {
-        me.updateRequiredMetadata(field.id, field.value);
-      }
+        if (angular.isDefined(me.requiredMetadata[field.id])) {
+          me.updateRequiredMetadata(field.id, field.value);
+        }
 
-      me.ud[target].fields[field.tabindex - 1] = field;
-    };
+        me.ud[target].fields[field.tabindex - 1] = field;
+      };
 
-    this.reset = function () {
-      me.ud = {};
-      me.metadata = NewEventMetadataResource.get(me.postProcessMetadata);
-    };
+      this.reset = function () {
+        me.ud = {};
+        me.metadata = NewEventMetadataResource.get(me.postProcessMetadata);
+      };
 
-    this.getFiledCatalogs = function () {
-      var catalogs = [];
+      this.getFiledCatalogs = function () {
+        var catalogs = [];
 
-      angular.forEach(me.ud, function(catalog) {
-        var empty = true;
-        angular.forEach(catalog.fields, function (field) {
-          if (angular.isDefined(field.presentableValue) && field.presentableValue !== '') {
-            empty = false;
+        angular.forEach(me.ud, function(catalog) {
+          var empty = true;
+          angular.forEach(catalog.fields, function (field) {
+            if (angular.isDefined(field.presentableValue) && field.presentableValue !== '') {
+              empty = false;
+            }
+          });
+
+          if (!empty) {
+            catalogs.push(catalog);
           }
         });
 
-        if (!empty) {
-          catalogs.push(catalog);
-        }
-      });
+        return catalogs;
+      };
 
-      return catalogs;
-    };
-
-    this.getUserEntries = function () {
-      angular.forEach(me.ud, function(catalog) {
-        catalog.empty = true;
-        angular.forEach(catalog.fields, function (field) {
-          if (angular.isDefined(field.presentableValue) && field.presentableValue !== '') {
-            catalog.empty = false;
-          }
+      this.getUserEntries = function () {
+        angular.forEach(me.ud, function(catalog) {
+          catalog.empty = true;
+          angular.forEach(catalog.fields, function (field) {
+            if (angular.isDefined(field.presentableValue) && field.presentableValue !== '') {
+              catalog.empty = false;
+            }
+          });
         });
-      });
 
-      return me.ud;
+        return me.ud;
+      };
     };
-  };
 
-  return new MetadataExtended();
-}]);
+    return new MetadataExtended();
+  }]);

--- a/modules/admin-ui-frontend/app/scripts/shared/services/wizards/new-series/metadataExtended.js
+++ b/modules/admin-ui-frontend/app/scripts/shared/services/wizards/new-series/metadataExtended.js
@@ -21,123 +21,125 @@
 'use strict';
 
 angular.module('adminNg.services')
-.factory('NewSeriesMetadataExtended', ['NewSeriesMetadataResource', function (NewSeriesMetadataResource) {
-  var MetadataExtended = function () {
-    var me = this, i;
+.factory('NewSeriesMetadataExtended', ['NewSeriesMetadataResource',
+  function (NewSeriesMetadataResource) {
+    var MetadataExtended = function () {
+      var me = this, i;
 
-    this.updateRequiredMetadata = function(fieldId, value) {
-      if (angular.isDefined(value) && value.length > 0) {
-        me.requiredMetadata[fieldId] = true;
-      } else {
-        me.requiredMetadata[fieldId] = false;
-      }
-    };
-
-    // As soon as the required metadata fields arrive from the backend,
-    // we check which are mandatory.
-    // This information will be needed in order to tell if we can move
-    // on to the next page of the wizard.
-    this.postProcessMetadata = function (data) {
-      var fields = [], chunk;
-      for (chunk in data) {
-        if (Object.prototype.hasOwnProperty.call(data, chunk)) {
-          // extended metadata is every object in the returned data which
-          // does not start with a dollar sign and which isn't dublincore/episode
-          if (chunk !== 'dublincore/series' && chunk.charAt(0) !== '$') {
-            me.ud[chunk] = {fields: data[chunk].fields};
-            me.ud[chunk].flavor = data[chunk].flavor;
-            me.ud[chunk].title = data[chunk].title;
-            fields = fields.concat(data[chunk].fields);
-          }
+      this.updateRequiredMetadata = function(fieldId, value) {
+        if (angular.isDefined(value) && value.length > 0) {
+          me.requiredMetadata[fieldId] = true;
+        } else {
+          me.requiredMetadata[fieldId] = false;
         }
-      }
-      // we go for the extended metadata here
-      if (fields.length > 0) {
-        for (i = 0; i < fields.length; i++) {
-          // just hooking the tab index up here, as this is already running through all elements
-          fields[i].tabindex = i + 1;
-          if (fields[i].required) {
-            me.updateRequiredMetadata(fields[i].id, fields[i].value);
-            if (fields[i].type === 'boolean') {
-              // set all boolean fields to false by default
-              fields[i].value = false;
+      };
+
+      // As soon as the required metadata fields arrive from the backend,
+      // we check which are mandatory.
+      // This information will be needed in order to tell if we can move
+      // on to the next page of the wizard.
+      this.postProcessMetadata = function (data) {
+        var fields = [], chunk;
+        for (chunk in data) {
+          if (Object.prototype.hasOwnProperty.call(data, chunk)) {
+            // extended metadata is every object in the returned data which
+            // does not start with a dollar sign and which isn't dublincore/episode
+            if (chunk !== 'dublincore/series' && chunk.charAt(0) !== '$') {
+              me.ud[chunk] = {fields: data[chunk].fields};
+              me.ud[chunk].flavor = data[chunk].flavor;
+              me.ud[chunk].title = data[chunk].title;
+              fields = fields.concat(data[chunk].fields);
             }
           }
         }
-        me.visible = true;
-      }
-      else {
-        me.visible = false;
-      }
-    };
-
-    // Checks if the current state of this wizard is valid and we are
-    // ready to move on.
-    this.isValid = function () {
-      var result = true;
-      //FIXME: The angular validation should rather be used,
-      // unfortunately it didn't work in this context.
-      angular.forEach(me.requiredMetadata, function (item) {
-        if (item === false) {
-          result = false;
+        // we go for the extended metadata here
+        if (fields.length > 0) {
+          for (i = 0; i < fields.length; i++) {
+            // just hooking the tab index up here, as this is already running through all elements
+            fields[i].tabindex = i + 1;
+            if (fields[i].required) {
+              me.updateRequiredMetadata(fields[i].id, fields[i].value);
+              if (fields[i].type === 'boolean') {
+                // set all boolean fields to false by default
+                fields[i].value = false;
+              }
+            }
+          }
+          me.visible = true;
         }
-      });
-      return result;
-    };
+        else {
+          me.visible = false;
+        }
+      };
 
-    this.save = function (scope) {
-      //FIXME: This should be nicer, rather propagate the id and values instead of looking for them in the parent scope.
-      var field = scope.$parent.params,
-          target = scope.$parent.target;
+      // Checks if the current state of this wizard is valid and we are
+      // ready to move on.
+      this.isValid = function () {
+        var result = true;
+        //FIXME: The angular validation should rather be used,
+        // unfortunately it didn't work in this context.
+        angular.forEach(me.requiredMetadata, function (item) {
+          if (item === false) {
+            result = false;
+          }
+        });
+        return result;
+      };
 
-      if (field.collection) {
-        if (angular.isArray(field.value)) {
-          field.presentableValue = field.value;
+      this.save = function (scope) {
+        //FIXME: This should be nicer,
+        //rather propagate the id and values instead of looking for them in the parent scope.
+        var field = scope.$parent.params,
+            target = scope.$parent.target;
+
+        if (field.collection) {
+          if (angular.isArray(field.value)) {
+            field.presentableValue = field.value;
+          } else {
+            field.presentableValue = field.collection[field.value];
+          }
         } else {
-          field.presentableValue = field.collection[field.value];
+          field.presentableValue = field.value;
         }
-      } else {
-        field.presentableValue = field.value;
-      }
 
-      if (angular.isDefined(me.requiredMetadata[field.id])) {
-        me.updateRequiredMetadata(field.id, field.value);
-      }
+        if (angular.isDefined(me.requiredMetadata[field.id])) {
+          me.updateRequiredMetadata(field.id, field.value);
+        }
 
-      me.ud[target].fields[field.tabindex - 1] = field;
-    };
+        me.ud[target].fields[field.tabindex - 1] = field;
+      };
 
-    this.getFiledCatalogs = function () {
-      var catalogs = [];
+      this.getFiledCatalogs = function () {
+        var catalogs = [];
 
-      angular.forEach(me.ud, function(catalog) {
-        var empty = true;
-        angular.forEach(catalog.fields, function (field) {
-          if (angular.isDefined(field.presentableValue) && field.presentableValue !== '') {
-            empty = false;
+        angular.forEach(me.ud, function(catalog) {
+          var empty = true;
+          angular.forEach(catalog.fields, function (field) {
+            if (angular.isDefined(field.presentableValue) && field.presentableValue !== '') {
+              empty = false;
+            }
+          });
+
+          if (!empty) {
+            catalogs.push(catalog);
           }
         });
 
-        if (!empty) {
-          catalogs.push(catalog);
-        }
-      });
+        return catalogs;
+      };
 
-      return catalogs;
+      this.reset = function () {
+        me.ud = {};
+        me.requiredMetadata = {};
+        me.metadata = NewSeriesMetadataResource.get(me.postProcessMetadata);
+      };
+
+      this.getUserEntries = function () {
+        return me.ud;
+      };
+
+      this.reset();
     };
 
-    this.reset = function () {
-      me.ud = {};
-      me.requiredMetadata = {};
-      me.metadata = NewSeriesMetadataResource.get(me.postProcessMetadata);
-    };
-
-    this.getUserEntries = function () {
-      return me.ud;
-    };
-
-    this.reset();
-  };
-
-  return new MetadataExtended();
-}]);
+    return new MetadataExtended();
+  }]);

